makefile

```diff
@@ -1,5 +1,6 @@
 #===============================================================================
```

Comment:
It would be nice to have this file with architecture-specific parts split out into their own files. That requires quite a bit of a refactor of this file so that we can keep as much common code in here as possible. It should be possible to just split out the setting of variables into architecture specific parts at the start of the file, and then use these later on in the makefile. I'm happy to have a look at this, if required.

Reply:
@keeranroth yup please have a look at this and do make suggestions as needed, this seems a bit tricky to isolate the parts.

Reply:
Ack. I'll start having a look at this tomorrow.